ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactionsMessageSender.java

```diff
@@ -37,16 +37,17 @@ public void sendTransactionsToPeers() {
   }
 
   private void sendTransactionsToPeer(final EthPeer peer) {
-    final Set<Hash> allTxToSend = transactionTracker.claimTransactionsToSendToPeer(peer);
-    while (!allTxToSend.isEmpty()) {
-      final LimitedNewPooledTransactionHashesMessages limitedTransactionsMessages =
-          LimitedNewPooledTransactionHashesMessages.createLimited(allTxToSend);
-      allTxToSend.removeAll(limitedTransactionsMessages.getIncludedTransactions());
-      try {
-        peer.send(limitedTransactionsMessages.getTransactionsMessage());
-      } catch (final PeerNotConnected e) {
-        return;
-      }
-    }
+    Iterables.partition(
+            transactionTracker.claimTransactionsToSendToPeer(peer),
+            4096 // implementation determined limit for how many hashes to send at once
+            )
+        .forEach(
+            hashes -> {
+              try {
+                peer.send(NewPooledTransactionHashesMessage.create(hashes));
+              } catch (final PeerNotConnected __) {
+                // if the peer isn't connected anymore, don't do anything
```

Comment:
the original code would return early if the peer was disconnected.
Do we want to check if the peer is still connected before sending?